mewarpx/mewarpx/utils_store/runsummary.py

```diff
@@ -0,0 +1,38 @@
+"""
+Utility class for printint out performance information after a run
+"""
+
+import time
+
+class RunSummary(object):
+    def __init__(self):
+        from pywarpx import callbacks
+        self.start_time = time.time()
+        self.prev_time = self.start_time
+        self.particle_steps_total = 0
+        self.total_particle_step_rate = 0
+        self.previous_particle_steps_total = 0
+        callbacks.installafterstep(self.update_particle_steps)
+
+    def update_particle_steps(self):
+        from mewarpx.mwxrun import mwxrun
```

Comment:
Same comment as above and also for line 23 below.

Reply:
The `mwxrun` import does have to happen here or else it throws an error about a circular import.

Reply:
I'm not familiar with the performance impact of imports but since this happens at every step it is worth considering. Do you know if importing `mwxrun` at every step would cause a slow-down?

Reply:
I did look it up and found a [stackoverflow post](https://stackoverflow.com/questions/32151193/is-there-a-performance-cost-putting-python-imports-inside-functions) that mentioned it. If I understand it right it there's no extra cost to just the `import` statement. Although it would be faster if I could figure out how to import just the one function that is needed. 

> For example, if you import math and then every time you need to use the sin(...) function you have to do math.sin(...), this will generally be slower than doing from math import sin and using sin(...) directly as the system does not have to keep looking up the function name within the module.

Reply:
I'm struggling to import the function directly like that though

Reply:
From what I understand, the python interpreter caches imported packages, so it does not cost extra to import packages that have already been imported, as it will just use the cached package that was imported previously.

Reply:
> From what I understand, the python interpreter caches imported packages, so it does not cost extra to import packages that have already been imported, as it will just use the cached package that was imported previously.

Yep nice that's exactly what the StackOverflow post says

Reply:
https://stackoverflow.com/questions/7336802/how-to-avoid-circular-imports-in-python 
This post gives some explanation about the issue. It may have to do with using from x import y, so you could try `import mewarpx.mwxrun` and then call `mewarpx.mwxrun.mwxrun` or something similar to avoid using the from .. import .. syntax. This will probably take some fiddling around with, but there's probably a way to bypass the circular import error without having to import in the function.